File: Data Structures & Algorithms/single-number/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    singleNumber(nums) {
        // 011
        // 011
        // 010
 
        // 010<- OR  010 <-AND 010 <- XOR
        // 011       011       011 
        // 011       010       001

        let x = 0
        for (const num of nums) {
            x ^= num
        }
        return x
    }
}

File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-0.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        heap = []
        for num in nums:
            if len(heap) >= k:
                currMax = heap[0]
                if num > currMax:
                    heapq.heapreplace(heap, num)
            else:
                heapq.heappush(heap, num)
        
        return heap[0]
        
